compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -1776,8 +1780,11 @@ abstract class GenJSCode extends plugins.PluginComponent
       val sym = fun.symbol
 
       def isRawJSDefaultParam: Boolean = {
-        sym.hasFlag(reflect.internal.Flags.DEFAULTPARAM) &&
-        (isRawJSType(sym.owner.tpe) || isRawJSCtorDefaultParam(sym))
+         isRawJSCtorDefaultParam(sym) || (
+             sym.hasFlag(reflect.internal.Flags.DEFAULTPARAM) &&
+             !isCtorDefaultParam(sym) &&
+             isRawJSType(sym.owner.tpe)
+         )
```

Comment:
I suggest to rewrite this with the `isCtorDefaultParam(sym)` test as a first thing:
```scala
if (isCtorDefaultParam(sym)) {
  isRawJSCtorDefaultParam(sym)
} else {
  sym.hasFlag(reflect.internal.Flags.DEFAULTPARAM) &&
  isRawJSType(sym.owner.tpe)
}
```